### PR TITLE
feat: add get service-cidr cmd to vcluster cli

### PR DESCRIPTION
**Pull request by pratikjagrut**
Adds `get service-cidr` command to vcluster cli.
```
vcluster get service-cidr
10.96.0.0/12
```

### CODE REVIEW

File: cmd/vclusterctl/cmd/get/service_cidr.go [hunk 1]
```diff
@@ -0,0 +1,78 @@
...
+		Namespace: globalFlags.Namespace,
+		log:       log.GetInstance(),
+	}
+
+	cobraCmd := &cobra.Command{
+		Use:   "service-cidr",
+		Short: "Prints Service CIDR of the cluster",
+		Long: `
+#######################################################
+############### vcluster get service-cidr  ###################
```
Comment (FabianKramm) [inline cmd/vclusterctl/cmd/get/service_cidr.go:32]: Please remove the extra `#####` at the end

File: cmd/vclusterctl/cmd/get/service_cidr.go [hunk 2]
```diff
@@ -0,0 +1,78 @@
...
+type serviceCIDRCmd struct {
+	Context   string
+	Namespace string
+	log       log.Logger
+}
+
+func getServiceCIDR(globalFlags *flags.GlobalFlags) *cobra.Command {
+	cmd := &serviceCIDRCmd{
+		Context:   globalFlags.Context,
+		Namespace: globalFlags.Namespace,
```
Comment (FabianKramm) [inline cmd/vclusterctl/cmd/get/service_cidr.go:23]: Please use the global flags here like in the other commands:
```
cmd := & serviceCIDRCmd{
		GlobalFlags: globalFlags,
		log:         log.GetInstance(),
	}
```

File: cmd/vclusterctl/cmd/get/service_cidr.go [hunk 3]
```diff
@@ -0,0 +1,78 @@
...
+	"github.com/loft-sh/vcluster/cmd/vclusterctl/flags"
+	"github.com/loft-sh/vcluster/cmd/vclusterctl/log"
+	"github.com/loft-sh/vcluster/pkg/helm/values"
+	"github.com/spf13/cobra"
+	"k8s.io/client-go/kubernetes"
+	"k8s.io/client-go/tools/clientcmd"
+)
+
+type serviceCIDRCmd struct {
+	Context   string
```
Comment (FabianKramm) [inline cmd/vclusterctl/cmd/get/service_cidr.go:15]: Embbed global flags here and remove context and namespace

File: cmd/vclusterctl/cmd/get/service_cidr.go [hunk 4]
```diff
@@ -0,0 +1,78 @@
...
+	if err != nil {
+		return err
+	}
+
+	cidr, err := values.GetServiceCIDR(client, "default", false)
+	if err != nil {
+		return err
+	}
+
+	output := "Service CIDR: " + cidr + "\n"
```
Comment (FabianKramm) [inline cmd/vclusterctl/cmd/get/service_cidr.go:71]: just print the service cidr, which makes it easier to use in an automated fashion

File: cmd/vclusterctl/cmd/get/service_cidr.go [hunk 5]
```diff
@@ -0,0 +1,78 @@
...
+	_, err = log.GetInstance().Write([]byte(output))
```
Comment (FabianKramm) [inline cmd/vclusterctl/cmd/get/service_cidr.go:72]: either write to os.Stdout or use `cmd.log.Write` here

File: cmd/vclusterctl/cmd/get/service_cidr.go [hunk 6]
```diff
@@ -0,0 +1,78 @@
...
+	if err != nil {
+		return fmt.Errorf("there is an error loading your current kube config (%v), please make sure you have access to a kubernetes cluster and the command `kubectl get namespaces` is working", err)
+	}
+
+	client, err := kubernetes.NewForConfig(kubeConfig)
+	if err != nil {
+		return err
+	}
+
+	cidr, err := values.GetServiceCIDR(client, "default", false)
```
Comment (FabianKramm) [inline cmd/vclusterctl/cmd/get/service_cidr.go:66]: use `cmd.Namespace` here

**Review by FabianKramm — CHANGES_REQUESTED** [review]
@pratikjagrut thanks for the PR! Please see my comments for changes. Would be great if you could also add that it works with ipv6 like the create command does

File: cmd/vclusterctl/cmd/get/service_cidr.go (hunk 1 above)
Comment (pratikjagrut) [reply cmd/vclusterctl/cmd/get/service_cidr.go]: done

File: cmd/vclusterctl/cmd/get/service_cidr.go (hunk 2 above)
Comment (pratikjagrut) [reply cmd/vclusterctl/cmd/get/service_cidr.go]: done

File: cmd/vclusterctl/cmd/get/service_cidr.go (hunk 3 above)
Comment (pratikjagrut) [reply cmd/vclusterctl/cmd/get/service_cidr.go]: done

File: cmd/vclusterctl/cmd/get/service_cidr.go (hunk 6 above)
Comment (pratikjagrut) [reply cmd/vclusterctl/cmd/get/service_cidr.go]: done

File: cmd/vclusterctl/cmd/get/service_cidr.go (hunk 4 above)
Comment (pratikjagrut) [reply cmd/vclusterctl/cmd/get/service_cidr.go]: done

File: cmd/vclusterctl/cmd/get/service_cidr.go [hunk 7]
```diff
@@ -0,0 +1,121 @@
...
+			}, metav1.CreateOptions{})
+			if err != nil {
+				return errors.Wrap(err, "create namespace")
+			}
+		} else if !kerrors.IsForbidden(err) {
+			return err
+		}
+	}
+
+	cidr, err := values.GetServiceCIDR(client, cmd.Namespace, false)
```
Comment (FabianKramm) [inline cmd/vclusterctl/cmd/get/service_cidr.go:102]: Would be great if we would avoid code duplication and make this a function both this command and create uses.

File: cmd/vclusterctl/cmd/get/service_cidr.go [hunk 8]
```diff
@@ -0,0 +1,121 @@
...
+		cmd.Namespace, _, err = kubeClientConfig.Namespace()
+		if err != nil {
+			return err
+		} else if cmd.Namespace == "" {
+			cmd.Namespace = "default"
+		}
+	}
+
+	// make sure namespace exists
+	_, err = client.CoreV1().Namespaces().Get(context.Background(), cmd.Namespace, metav1.GetOptions{})
```
Comment (FabianKramm) [inline cmd/vclusterctl/cmd/get/service_cidr.go:84]: we don't want to enforce this here, as creating a namespace shouldn't be part of this command, so just delete this block

File: cmd/vclusterctl/cmd/get/service_cidr.go [hunk 9]
```diff
@@ -0,0 +1,121 @@
...
+		if idx != -1 || idz != -1 {
+			cidr, err = values.GetServiceCIDR(client, cmd.Namespace, true)
+		}
+		if err != nil {
+			cmd.log.Warn(err)
+			cidr = "10.96.0.0/12"
+		}
+	}
+
+	_, err = cmd.log.Write([]byte(cidr + "\n"))
```
Comment (FabianKramm) [inline cmd/vclusterctl/cmd/get/service_cidr.go:115]: just print the cidr, no newline required

**Review by FabianKramm — CHANGES_REQUESTED** [review]
@pratikjagrut thanks for the update, please see my comments

File: cmd/vclusterctl/cmd/get/service_cidr.go (hunk 7 above)
Comment (pratikjagrut) [reply cmd/vclusterctl/cmd/get/service_cidr.go]: done

**Review by FabianKramm — APPROVED** [review]
Thanks, LGTM!

**Pull request merged**